backend/webserver/wittymail_server/emailapi.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText 
+from email.mime.base import MIMEBase 
+from email import encoders 
+import os
+import util.logger as logger
+
+log = logger.get_logger(__name__)
+
+# Boolean to make sure we do not try to send an email before login
+INITIALIZED = False
+
+def set_login_details(server, port, username, password):
+  assert server is not None and port is not None and username is not None and password is not None,\
+  log.error('server, port number, username or password is None')
+
+  # s is a global object used to interact with the SMTP server
+  global s
+
+  # Connect to the SMTP server and login
+  try:
+    s = smtplib.SMTP(server, port)
+    s.ehlo()
+    s.starttls() 
+    s.login(username, password) 
+
+  except Exception as e:
+     log.error('Incorrect server/user information provided')
+     return [-1, 'Incorrect server/user information provided']
+
+  global INITIALIZED
+  INITIALIZED = True
+  log.debug('Login successful')
+  return [0, 'Login successful']
+
+def send_email(frm, to, subject, body, cc = None, attachment = None):
+  assert                  \
+  frm is not None and     \
+  to is not None and      \
+  subject is not None and  \
+  body is not None, log.error('Incorrect value provided from=%s to=%s subject=%s body=%s'  
```

Comment:
Need to return from fn here

Reply:
The assert will stop the execution anyway